### PR TITLE
feat: aws validator terraform example

### PR DESCRIPTION
**Pull request by paulbalaji**
Resolves https://github.com/hyperlane-xyz/issues/issues/987

Example terraform to run a Hyperlane Validator on AWS.

Includes:
- Main configuration to setup core infra.
- Modules for individually configuring:
	- IAM users, roles and policies.
	- S3 bucket for validator signatures, and its permissions.
	- Validator task on ECS, including any permissions and EFS volumes to be mounted.
- Script for generating the top-level terraform outputs, given the main configuration.

TODO:
- [x] add more inline comments
- [ ] update readme link (only after merging https://github.com/hyperlane-xyz/v3-docs/pull/45)

### CODE REVIEW

**Comment by avious00** [comment]
@paulbalaji sorry if i missed this, but will the terraform option also be in the aws guide content?

**Comment by paulbalaji** [comment]
> @paulbalaji sorry if i missed this, but will the terraform option also be in the aws guide content?

yup! The readme in this PR was copied over from https://github.com/hyperlane-xyz/v3-docs/pull/45

File: rust/terraform/README.md
Comment (paulbalaji) [inline rust/terraform/README.md:7]: can update later after merging https://github.com/hyperlane-xyz/v3-docs/pull/45

**Comment by avious00** [comment]
will let other folks verify the technical aspect, but i appreciate the context/readability you added in the inline comments. bullish

File: rust/terraform/modules/validator/variables.tf
Comment (tkporter) [inline rust/terraform/modules/validator/variables.tf:43]: fwiw we can definitely run on less than 6gb but I guess no harm in 6gb, i'm sure it's not very expensive

**Review by tkporter — APPROVED** [review]
Great job! This is super thorough and very easy to read. Super excited to see this happen. Approving pending some questions

How does it work to SSH into one of these? Also just wanna make sure that it's appropriately locked down

File: rust/terraform/generate-outputs.py
Comment (paulbalaji) [reply rust/terraform/generate-outputs.py:1]: Yup that's correct, will add a comment

File: rust/terraform/outputs.tf
Comment (paulbalaji) [reply rust/terraform/outputs.tf:2]: I think there's some low hanging fruit here to make it even easier to write outputs for each validator module without needing a script

I'll see if I can get that working and if so we can nuke this python

File: rust/terraform/generate-outputs.py
Comment (paulbalaji) [reply rust/terraform/generate-outputs.py:42]: I'll replace the logs to say that the outputs file has been updated, instead of printing it all

File: rust/terraform/modules/s3/main.tf
Comment (paulbalaji) [reply rust/terraform/modules/s3/main.tf:66]: Stale description, validator logs are only viewable on cloudwatch

File: rust/terraform/modules/validator/variables.tf
Comment (paulbalaji) [reply rust/terraform/modules/validator/variables.tf:43]: Lemme know what some good numbers are, I used 6gb because it's at like 80% utilisation

Comment (tkporter) [reply rust/terraform/modules/validator/variables.tf:43]: ah just leave as 6gb in that case

File: rust/terraform/modules/s3/main.tf
Comment (paulbalaji) [reply rust/terraform/modules/s3/main.tf:64]: good catch actually - lemme try putting it all under the bucket policy

File: rust/terraform/globals.tf
Comment (paulbalaji) [reply rust/terraform/globals.tf:82]: the validator doesn't have a public IP, it's still restricted to those who can access the VPC

**Comment by paulbalaji** [comment]
> How does it work to SSH into one of these? Also just wanna make sure that it's appropriately locked down

Yup it's locked down. You can use [session manager](https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager.html) to ssh into an agent, see fargate references [here](https://aws.amazon.com/blogs/containers/new-using-amazon-ecs-exec-access-your-containers-fargate-ec2/)

**Pull request merged**
